logicrunner/logicrunner.go

```diff
@@ -614,35 +615,72 @@ func (lr *LogicRunner) OnPulse(ctx context.Context, pulse core.Pulse) error {
 
 	// send copy for validation
 	for ref, state := range lr.state {
+		state.Lock()
+
+		// some old stuff
 		state.RefreshConsensus()
 
 		if es := state.ExecutionState; es != nil {
+			es.Lock()
+
+			es.pending = es.pending || es.Current != nil
+			queue := es.releaseQueue()
+
 			caseBind := es.Behaviour.(*ValidationSaver).caseBind
+			requests := caseBind.getCaseBindRequestsFromMessageBus(ctx)
+
 			messages = append(
 				messages,
-				caseBind.ToValidateMessage(ctx, ref, pulse),
-				caseBind.ToExecutorResultsMessage(ref),
+				&message.ValidateCaseBind{
+					RecordRef: ref,
+					Requests:  requests,
+					Pulse:     pulse,
+				},
+				&message.ExecutorResults{
+					RecordRef: ref,
+					Pending:   state.ExecutionState.pending,
+					Requests:  requests,
+					Queue:     convertQueueToMessageQueue(queue),
+				},
 			)
 
-			es.ReleaseQueue()
-
 			// TODO: if Current is not nil then we should request here for a delegation token
 			// to continue execution of the current request
```

Comment:
By any chance doesn't this PR make this comment obsolete?

Reply:
no, we don't have delegation logic yet